File: 프로그래머스/3/43162. 네트워크/네트워크.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var count: Int = 0
    var visited: [Bool] = Array(repeating: false, count: n)
    
    func dfs(_ index: Int) {
        let currentNetwork = computers[index]
        
        for i in 0..<n {
            if currentNetwork[i] == 1 && !visited[i] {
                visited[i] = true
                dfs(i)
            }
        }
    }
    
    for i in 0..<n {
        if !visited[i] {
            visited[i] = true
            dfs(i)
            count += 1
        }
    }
    
    return count
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = new Set()
    
    for (let i=0; i<n; i++) {
        // 방문한 컴퓨터라면 다음으로
        if (visited.has(i)) continue;
        visited.add(i)
        answer++
        dfs(computers, visited, i)
    }
    
    return answer;
}

function dfs(computers, visited, i) {
    const adjComputers = computers[i]
    
    for (const [index, value] of adjComputers.entries()) {
        // 방문 했다면 다음으로
        if (visited.has(index) || value === 0) continue;
        // 방문처리하고 재귀호출
        visited.add(index)
        dfs(computers, visited, index)
    }
}